### ISSUE TITLE
ArchLinux any not recognized in platforms

### ISSUE THREAD

**Issue by wookietreiber** [ansible/ansible-lint#3049]
##### Summary

ansible-lint does not recognize [ArchLinux any platform](https://galaxy.ansible.com/api/v1/platforms/?search=ArchLinux) in `meta/main.yml`.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 6.13.1.dev0 using ansible 2.14.2
You are using a pre-release version of ansible-lint.
```

- ansible installation method: OS package
- ansible-lint installation method: OS package

##### STEPS TO REPRODUCE

```bash
git clone https://github.com/idiv-biodiversity/ansible-role-ssh-audited.git
cd ansible-role-ssh-audited/
git checkout v0.0.4
ansible-lint
```

##### Desired Behavior

ansible-lint should not complain as ArchLinux any is in platforms list [here](https://galaxy.ansible.com/api/v1/platforms/?search=ArchLinux). ansible-lint didn't complain before about having ArchLinux any in platforms. 

##### Actual Behavior

```
WARNING  Listing 1 violation(s) that are fatal
schema[meta]: {'name': 'ArchLinux', 'versions': ['any']} is not valid under any of the given schemas
meta/main.yml:1  Returned errors will not include exact line numbers, but they will mention
the schema name being used as a tag, like ``schema[playbook]``,
``schema[tasks]``.

This rule is not skippable and stops further processing of the file.

If incorrect schema was picked, you might want to either:

* move the file to standard location, so its file is detected correctly.
* use ``kinds:`` option in linter config to help it pick correct file type.


Read documentation for instructions on how to ignore specific rule violations.

             Rule Violation Summary
 count tag          profile rule associated tags
     1 schema[meta] basic   core

Failed after min profile: 1 failure(s), 0 warning(s) on 7 files.
```

**Comment by ssbarnea** [comment]
Official documentation only mentioned "all" not "any", so the bug was... nearby.

**Comment by wookietreiber** [comment]
I can confirm that both removing `versions:` and `all` don't show an error. Thanks! 

> Official documentation only mentioned "all" not "any", so the bug was... nearby.

Where is that mentioned? I only ever looked at the galaxy platforms API [here](https://galaxy.ansible.com/api/v1/platforms/?search=ArchLinux). Shouldn't this be changed, then? 

Also, is it possible to improve the error message to say something like `$PLATFORM is a supported platform, supported versions are $LIST or check at $LINK for supported versions`. IMO this would avoid confusion, especially since `any` was valid before for ArchLinux. The current error message does not help in figuring that out.

**Comment by ssbarnea** [comment]
The current error message are coming from python-jsonschema library, so feel free to file a bug report there. Short answer is that what we can do with schema is very limited.

Historically Galaxy allowed upload with random strings as versions, so in that particular case "any" acts as a specific version and not as an "all" wildcard version.

I will notify Galaxy team about this and see if there is a something we can to to improve the situation.
